src/de/tuebingen/uni/ub/hc/MARCProcessing/MARC4JProcessor.java

```diff
@@ -0,0 +1,169 @@
+                ControlField curControlField = (ControlField) record.getVariableField("008");
+                data = curControlField.getData();
+
+                // the three-character MARC language code takes character
+                // positions 35-37
+               lang = data.substring(35, 38);
+                
+                // create small corpus for debugging purposes
+                // if(lang.contains("ger")){
+                //// System.out.println(lang+" "+countRecForTest);
+                // writer.write(record);
+                // countRecForTest +=1;
+                // }
+
+                // get author 101
+                currentField = (DataField) record.getVariableField("100");
+
+                // System.out.println(record.toString());
+                if (currentField != null) {
+                    subfield = currentField.getSubfield('a');
+                    if (subfield != null) {
+                        author = subfield.getData();
+                    }
+                    subfield = currentField.getSubfield('0');
+                    if (subfield != null) {
+                        authorGND = subfield.getData();
+                    }
+                }
+                // get second author
+                currentField = (DataField) record.getVariableField("700");
+
+                // System.out.println(record.toString());
+                if (currentField != null) {
+                    subfield = currentField.getSubfield('a');
+                    if (subfield != null) {
+                       secAuthor = subfield.getData();
+                    }
+                    subfield = currentField.getSubfield('0');
+                    if (subfield != null) {
+                        secAuthorGND = subfield.getData();
+                    }
+                }
+
+                // get data field 245 title
+                currentField = (DataField) record.getVariableField("245");
+                // System.out.println(currentField);
+                // get the title proper
+                subfield = currentField.getSubfield('a');
+                title = subfield.getData();
+                title = title.replaceAll("\\u0098", "");
+                subfield = currentField.getSubfield('b');
+                if (subfield != null) {
+                    subtitle = subfield.getData();
+                }
+                //This field contains a "Herausgeber" if no Author is named
+                subfield = currentField.getSubfield('c');
+                if (subfield != null && author.equalsIgnoreCase("0")) {
+                    author = subfield.getData();
+                }
+
+                // System.out.println(ixtRecord.getAuthor());
+                ixtRecord = new IxTheoRecord(ppn, lang, author, authorGND, secAuthor, secAuthorGND, title, subtitle, ixTheoAnnoSet);
+                // when record is filled add to corpus
+                corpus.getRecordList().add(ixtRecord);
```

Comment:
I would give the corpus class an addIxTheoRecord() method that takes the arguments for the IxTheoRecord constructor and then pass them through.  Maybe you don't even need the getRecordList() method at all.